basalt-server-lib/src/services/questions.rs

```diff
@@ -37,10 +52,7 @@ pub struct QuestionResponse {
 impl QuestionResponse {
     fn from(value: &Problem, languages: &LanguageSet, show_hidden: bool) -> Self {
         Self {
-            languages: value
-                .languages
-                .clone()
-                .unwrap_or_else(|| languages.iter().map(|l| l.name().to_string()).collect()),
+            languages: languages.iter().map(LanguageSyntax::from).collect(),
```

Comment:
This change makes it so that languages can no longer be specified for a question.  This needs to use the `value.languages` if it is Some.  This should only use `languages` if `value.languages` is none.

I think something like 
```suggestion
            languages: value
                .languages
                .clone()
                .map(|l| l.iter().map(LanguageSyntax::from).collect())
                .unwrap_or_else(|| languages.iter().map(LanguageSyntax::from).collect()),
```
may work